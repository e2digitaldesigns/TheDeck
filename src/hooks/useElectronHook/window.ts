/* istanbul ignore file */
const windowCopy: any = window;
export { windowCopy as window };
