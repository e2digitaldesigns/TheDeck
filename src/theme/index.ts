export * from "./styles/forms.styles";
export * from "./styles/scrollBars.styles";
