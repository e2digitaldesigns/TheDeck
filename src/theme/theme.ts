export const theme = {
  sizes: {
    application: {
      height: "785px",
      width: "1280px"
    },
    header: {
      height: "2rem",
      width: "100vw"
    },
    macroDeck: {
      sideBar: { width: "240px" },
      mainContent: { width: "1016px" }
    }
  },
  colors: {
    body: "black",
    border: "#111",
    accent1: "#f542e9",
    accent2: "#291697",
    accent3: "#404040",
    accent4: "#2b2b2b",
    accent5: "#1a1a1a",
    gridBg: "43, 45, 48,",
    font: "white",
    main: "#100727",
    mainBackground: "black",
    scrollBackground: "#100727",
    scrollFont: "#dedede",
    scrollHeader: "black",
    scrollHeaderFont: "white"
  }
};
