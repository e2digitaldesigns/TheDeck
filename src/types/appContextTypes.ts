export interface IntAppDataIconSelector {
  isVisible: boolean;
  icon: string;
}

export interface IntAppDataActive {
  profileId: string;
  pageId: string;
  buttonPadId: string;
  actionId: string;
}

export interface IntDropZones {
  actionList: boolean;
  buttonPads: boolean;
  sideBarProfiles: boolean;
  sideBarStyles: boolean;
  styleHeader: boolean;
}

export interface IntAppContextInterface {
  iconSelector: IntAppDataIconSelector;
  active: IntAppDataActive;
  dropZones: IntDropZones;
}

export interface IntAppContextStateInterface {
  appState: IntAppContextInterface;
  setAppState: React.Dispatch<React.SetStateAction<any>>;
}

export type IntAppData = IntAppContextStateInterface;

export enum AppStateActive {
  ProfileId = "profileId",
  PageId = "pageId",
  ButtonPadId = "buttonPadId",
  ActionId = "actionId"
}
