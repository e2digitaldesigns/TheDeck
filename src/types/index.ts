/* istanbul ignore file */
export * from "./actionType";
export * from "./aplicationTypes";
export * from "./appContextTypes";
export * from "./buttonPadGridTypes";
export * from "./dragAndDropTypes";
export * from "./globalContextType";
export * from "./globalType";
export * from "./ipcRenderTypes";
export * from "./obsTypes";
