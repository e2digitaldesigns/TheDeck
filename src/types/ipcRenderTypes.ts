export enum IpcRendererTypes {
  database = "database",
  databaseReturn = "database:return",
  header = "header",
  importSettings = "importSettings",
  importSettingsReturn = "importSettings:return",
  mdPlaySound = "MD:playSound"
}

export enum IpcRendererTypesAction {
  close = "close",
  fsToggle = "fsToggle",
  minimize = "minimize",
  loadAppData = "loadAppData",
  saveAppData = "saveAppData"
}
