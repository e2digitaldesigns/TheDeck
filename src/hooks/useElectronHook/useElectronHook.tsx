import {
  IntGlobalContextInterface,
  IpcRendererTypes,
  IpcRendererTypesAction
} from "./../../types";

import { window } from "./window";
// declare const window: any;

export interface IntElectronHook {
  ipcRender: (
    name: IpcRendererTypes,
    data: IpcRendererTypesAction | object
  ) => void;
  loadAppData: () => void;
  saveAppData: (data: IntGlobalContextInterface) => void;
}

const ipcRenderer = window?.electron?.ipcRenderer;

const useElectronHook = (): IntElectronHook => {
  const ipcRender = (name: string, data: string | object) => {
    ipcRenderer && ipcRenderer.send(name, data);
  };

  const loadAppData: IntElectronHook["loadAppData"] = () => {
    ipcRenderer &&
      ipcRenderer.send(IpcRendererTypes.database, {
        action: IpcRendererTypesAction.loadAppData
      });
  };

  const saveAppData: IntElectronHook["saveAppData"] = data => {
    ipcRenderer &&
      ipcRenderer.send(IpcRendererTypes.database, {
        action: IpcRendererTypesAction.saveAppData,
        data
      });
  };

  return {
    ipcRender,
    loadAppData,
    saveAppData
  };
};

export default useElectronHook;
