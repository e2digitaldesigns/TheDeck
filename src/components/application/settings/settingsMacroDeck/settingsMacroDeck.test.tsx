import "@testing-library/jest-dom";
import { window } from "../../../../hooks/useElectronHook/window";
import { createEvent, fireEvent, render } from "@testing-library/react";
import _cloneDeep from "lodash/cloneDeep";
import SettingsMacroDeck from "./settingsMacroDeck";
import { ThemeProvider } from "styled-components";
import { theme } from "../../../../theme/theme";

import initGlobalState from "../../../../__mock_data__/globalState.json";
import { IntGlobalData } from "../../../../types";

const mockGlobalState: IntGlobalData = {
  state: _cloneDeep(initGlobalState),
  setState: jest.fn(state => state)
};

const mocked = {
  ipcRender: jest.fn(),
  ipcRendererOn: jest.fn(),
  ipcRendererSender: jest.fn(),
  saveAs: jest.fn()
};

jest.mock("../../../../hooks/useElectronHook/window", () => ({
  window: {
    electron: {
      ipcRenderer: {
        on: () => mocked.ipcRendererOn(),
        removeListener: () => jest.fn(),
        send: (name: string, data: string | object) =>
          mocked.ipcRendererSender(name, data)
      }
    }
  }
}));

jest.mock("../../../../hooks", () => ({
  useAppData: () => ({ appState: { dropZones: { buttonPads: false } } }),
  useButton: () => ({
    activateButtonPad: jest.fn(),
    createButtonPad: jest.fn(),
    deleteButtonPad: jest.fn(),
    readButtonPad: () => mockGlobalState.state.buttonPads[0]
  }),
  useElectron: () => ({
    ipcRender: mocked.ipcRender
  }),
  useGlobalData: () => mockGlobalState,
  useProfile: () => ({
    readProfile: () => mockGlobalState.state.profiles[0]
  })
}));

jest.mock("file-saver", () => ({
  saveAs: () => mocked.saveAs()
}));

const setup = () => {
  return render(
    <ThemeProvider theme={theme}>
      <SettingsMacroDeck />
    </ThemeProvider>
  );
};

afterEach(() => {
  jest.clearAllMocks();
});

describe("<SettingsMacroDeck />", () => {
  it("Should render correctly", () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it("Should start IPC Render On", () => {
    setup();
    expect(mocked.ipcRendererOn).toHaveBeenCalledTimes(1);
  });

  it("Should export data", () => {
    const wrapper = setup();
    const buttonExport = wrapper.getByTestId(
      "settings_macro_deck__export-button"
    );
    expect(buttonExport).toBeTruthy();
    fireEvent.click(buttonExport);
    expect(mocked.saveAs).toHaveBeenCalledTimes(1);
  });

  it("Should import data", () => {
    const wrapper = setup();
    const filePath = "cool/and/the/gang.music";

    const inputFile = wrapper.getByTestId(
      "settings_macro_deck__import-input"
    ) as HTMLInputElement;
    expect(inputFile).toBeTruthy();

    const importBtn = wrapper.getByTestId("settings_macro_deck__import-button");
    expect(importBtn).toBeTruthy();

    fireEvent(
      inputFile,
      createEvent("input", inputFile, {
        target: { files: filePath }
      })
    );

    expect(inputFile.files).toBe(filePath);

    fireEvent.click(importBtn);
    expect(mocked.ipcRender).toHaveBeenCalledTimes(1);
    expect(mocked.ipcRender).toHaveBeenCalledWith("importSettings", {
      filePath: undefined,
      importType: "local"
    });
  });
});
