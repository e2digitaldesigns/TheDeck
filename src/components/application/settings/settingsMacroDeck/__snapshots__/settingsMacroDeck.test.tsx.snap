// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsMacroDeck /> Should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-pVTFL cScoAe"
      >
        <fieldset
          class="sc-jrQzAO fFPygj"
        >
          <button
            data-testid="settings_macro_deck__export-button"
          >
            Export
          </button>
          <br />
          <br />
          <hr />
          <br />
          <br />
          <input
            accept=".json"
            data-testid="settings_macro_deck__import-input"
            id="importSettingsFile"
            type="file"
          />
          <button
            data-testid="settings_macro_deck__import-button"
          >
            Import
          </button>
        </fieldset>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-pVTFL cScoAe"
    >
      <fieldset
        class="sc-jrQzAO fFPygj"
      >
        <button
          data-testid="settings_macro_deck__export-button"
        >
          Export
        </button>
        <br />
        <br />
        <hr />
        <br />
        <br />
        <input
          accept=".json"
          data-testid="settings_macro_deck__import-input"
          id="importSettingsFile"
          type="file"
        />
        <button
          data-testid="settings_macro_deck__import-button"
        >
          Import
        </button>
      </fieldset>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
